### DISCUSSION TITLE
New Beta Test: Docker Compose Plugin

### DISCUSSION THREAD

**Discussion by navarr** [wardenenv/warden#638] [Announcements]
Happy Friday, everyone!

Today, I've merged #631 into the `main` branch for beta testing.  This PR moves Warden from using the deprecated `docker-compose` standalone to the `docker compose` built-in plugin.

It is expected to resolve issues with pulling the MariaDB image and cause no major new issues.  That said, there is a small BC break to be aware of.

## Compatibility Issue

Moving to `docker compose` over the standalone means that image names will update, and any stored configuration for connecting to databases from the host machine will need updating.  This change will cause a db container for a project named vanilla to change from `vanilla_db_1` to `vanilla-db-1`.  Please keep this in mind if you cannot connect to an image after the update.  In all cases, connecting from one image to another in the network should still be possible through the use of the shortname (e.g. `db`).

## How to Test

To use the current in-development version of Warden you can either checkout the `main` branch if you installed via git, or you can run `brew install wardenenv/warden/warden --HEAD` to have brew upgrade to the in-development version.

## How to Report

Please report issues as normal through the [Warden New Issue Wizard](https://github.com/wardenenv/warden/issues/new/choose).  Please specify that you are using the `main` branch.

**Comment by tdgroot** [comment]
Really in favor of this change, but I do have one comment.

To allow for BC, I'd like to see a DOCKER_COMPOSE command variable to be used instead of hardcoded `docker compose` commands. This can be useful when trying out newer/older versions of Docker Compose or even having to use a separate Docker Compose binary when using Podman.

**Reply by navarr** [reply]
That's a really good thought, @tdgroot and I'll move forward with it before release!

Thank you!

**Comment by sprankhub** [comment]
Would it be possible to release this soonish, @navarr? Is there anything preventing this from happening? Thanks!

**Reply by navarr** [reply]
I wanted to release it with more as v1 but I can see it's a bit more important.

I'll release it this week since the beta has been up and nobody has reported issues with it. Though I'll have to double check that

**Reply by sprankhub** [reply]
Would be nice if this would be released ;-) Thanks!

**Comment by navarr** [comment]
Released as v0.14.0
